assignment+4/Main.java

```diff
@@ -5,152 +5,168 @@
 - Retrieve records matching given unique record ID
 - Retrieve records that fall within a given geographic region */
 
-package assignment4;
-import java.io.*;
-import java.util.*;
-
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
 public class Main {
 
-    // returns a BufferedReader object for a given file
-    public static BufferedReader fileReader(String filename) throws IOException {
-      File file = new File(filename);
-      FileReader fr = new FileReader(file);
-      BufferedReader br = new BufferedReader(fr);
-      return br;
-    }
+  public static int NUM_RECORDS = 10000; // globally available # of records count
+
+  // reads a given csv file line by line and returns an array of records
+  public static ArrayList<Record> readCSV(String filename) {
+    ArrayList<Record> records = new ArrayList<Record>();
+    BufferedReader br = null;
+    String line = null;
+    int counter = 0; // we'll use this counter to
+    try {
+      br = new BufferedReader(new FileReader(filename));
+      br.readLine(); // skip the first line, usually the header in CSV files
+      while ((line = br.readLine()) != null) {
+        // some logic to handle bad inputs from CSV, like "," in the geographicName
+        String[] input = line.split(",(?=([^\"]*\"[^\"]*\")*[^\"]*$)"); // -1 will account for missing location value
+        // build a record object based on CSV input
+        String cgndbID = null;
+        String geographicName = null;
+        String geographicTerm = null;
+        Double latitude = null;
+        Double longitude = null;
+        String location = null;
+        String province = null;
+
+        for(int i = 0; i < input.length; i++){
```

Comment:
what is this doing..

Reply:
reading each line in any given CSV and trying to parse the data, surrounded by a try/catch block for fault tolerance (so our program doesn't shit the bed when a null field is found)